src/components/settings/settings-knowledge-base/settings-knowledge-base.js

```diff
@@ -25,106 +28,125 @@ import NavigationModal from '../../navigation-modal';
-    intl: PropTypes.object.isRequired,
-    isCreateMode: PropTypes.bool,
-    kbCredentials: KbCredentials.KbCredentialsReduxStateShape,
-    match: ReactRouterPropTypes.history.isRequired,
-    onDelete: PropTypes.func.isRequired,
-    onSubmit: PropTypes.func.isRequired,
-  };
-
-  state = {
-    toasts: [],
-    deleteConfirmationModalDisplayed: false,
-  }
-
-  componentDidUpdate(prevProps) {
-    const {
-      config,
-      kbCredentials,
-      history,
-    } = this.props;
 
+const propTypes = {
+  config: KbCredentials.CredentialShape,
+  currentKBName: PropTypes.string.isRequired,
+  isCreateMode: PropTypes.bool,
+  kbCredentials: KbCredentials.KbCredentialsReduxStateShape,
+  kbId: PropTypes.string.isRequired,
+  onDelete: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  isCreateMode: false,
+};
+
+const SettingsKnowledgeBase = ({
+  config,
+  currentKBName,
+  isCreateMode,
+  kbCredentials,
+  kbId,
+  onDelete,
+  onSubmit,
+}) => {
+  const intl = useIntl();
+  const history = useHistory();
+
+  const [toasts, setToasts] = useState([]);
+  const [deleteConfirmationModalDisplayed, setDeleteConfirmationModalDisplayed] = useState(false);
+
+  useEffect(() => {
     if (kbCredentials.hasUpdated) {
       history.push({
         pathname: `/settings/eholdings/knowledge-base/${config.id}`,
-        state: { eholdings: true, isFreshlySaved: true }
+        state: {
+          eholdings: true,
+          isFreshlySaved: true,
+        },
       });
 
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState(({ toasts }) => ({
-        toasts: [...toasts, {
+      setToasts([
+        ...toasts,
+        {
           id: `settings-kb-${config.id}`,
           message: <FormattedMessage id="ui-eholdings.settings.kb.updated" />,
-          type: 'success'
-        }],
-      }));
+          type: 'success',
+        },
+      ]);
     }
 
     if (kbCredentials.hasSaved) {
       history.push({
         pathname: `/settings/eholdings/knowledge-base/${config.id}`,
-        state: { eholdings: true, isFreshlySaved: true }
+        state: {
+          eholdings: true,
+          isFreshlySaved: true,
+        },
       });
 
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState(({ toasts }) => ({
-        toasts: [...toasts, {
+      setToasts([
+        ...toasts,
+        {
           id: `settings-kb-${config.id}-${Date.now()}`,
-          message: <FormattedMessage id="ui-eholdings.settings.kb.saved" values={{ name: config.attributes.name }} />,
-          type: 'success'
-        }],
-      }));
+          message: (
+            <FormattedMessage
+              id="ui-eholdings.settings.kb.saved"
+              values={{ name: config.attributes.name }}
+            />
+          ),
+          type: 'success',
+        },
+      ]);
     }
+  }, [kbCredentials]);
+
+  const prevErrors = useRef(kbCredentials.errors).current;
```

Comment:
Isn't just using `useEffect` with `kbCredentials.errors` dependency enough to know that errors changed?

Reply:
No, it isn't. We need `prevProps` to make it work as expected. Without `useRef` and using just `useEffect` we will get this behavior. We could try to make another action to clear errors when component will destroyed, but I think that this is out of scope of this task. 
![HhsOxkQ4s3](https://user-images.githubusercontent.com/55701515/131491636-92acaa88-8602-473b-8122-f9dac8569739.gif)

Reply:
I see, ok, let's keep it this way